Libraries/pbxbuild/Sources/Tool/CopyResolver.cpp

```diff
@@ -77,13 +78,20 @@ ResolveInternal(
     Tool::OptionsResult options = Tool::OptionsResult::Create(toolEnvironment, toolContext->workingDirectory(), nullptr);
     Tool::Tokens::ToolExpansions tokens = Tool::Tokens::ExpandTool(toolEnvironment, options, std::string(), args);
 
+    std::vector<std::string> toolInputPaths = toolEnvironment.inputs(toolContext->workingDirectory());
+
     // TODO(grp): This should be generic for all tools.
     std::vector<Tool::Invocation::DependencyInfo> dependencyInfo;
     if (tool->deeplyStatInputDirectories()) {
         for (std::string const &inputPath : inputPaths) {
             /* Create a dependency info file to track the input directory contents. */
-            auto info = Tool::Invocation::DependencyInfo(dependency::DependencyInfoFormat::Directory, inputPath);
-            dependencyInfo.push_back(info);
+            libutil::DefaultFilesystem filesystem = libutil::DefaultFilesystem();
```

Comment:
It should be fine since the inputs are listed explicitly for Ninja. The directory dependency info tool is to record the contents of the directory, so Ninja notices if directory contents change, rather than just if the directory itself does. For files that's not important.